sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -36,9 +44,9 @@ export interface ConnectionContext extends ConnectionContextBase {
   readonly config: EventHubConnectionConfig;
   /**
    * The credential to be used for Authentication.
-   * Default value: SharedKeyCredentials.
+   * Default value: TokenProvider.
    */
-  tokenCredential: SharedKeyCredential | TokenCredential;
+  tokenCredential: TokenProvider | TokenCredential;
```

Comment:
At a quick glance, this does not tell me why I would need two different interfaces TokenProvider and TokenCredential, essentially both of which have a getToken() method. Shape wise, the difference is that one is async, the other is not. It would have been great to instead infer that one is for AAD, the other is for SAS which was the reason behind me suggesting that we use the SasTokenProvider class directly here though I understand your concerns around instanceOf checks

Can we rename the interface to SasTokenProvider and the class to SasTokenProviderImpl?

Reply:
That's fair. Will rename.

Reply:
Updated!